shell/platform/darwin/ios/ios_context_metal.mm

```diff
@@ -0,0 +1,84 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/ios_context_metal.h"
+
+#include "flutter/fml/logging.h"
+
+namespace flutter {
+
+IOSContextMetal::IOSContextMetal() {
+  device_.reset([MTLCreateSystemDefaultDevice() retain]);
+  if (!device_) {
+    FML_LOG(ERROR) << "Could not acquire Metal device.";
+    return;
+  }
+
+  main_queue_.reset([device_ newCommandQueue]);
+
+  if (!main_queue_) {
+    FML_LOG(ERROR) << "Could not create Metal command queue.";
+    return;
+  }
+
+  [main_queue_ setLabel:@"Flutter Main Queue"];
+
+  main_context_ = GrContext::MakeMetal([device_ retain], [main_queue_ retain]);
```

Comment:
This is non idiomatic, retaining in function calls.  Are you sure thats how those functions work?

Reply:
If it is correct, consider adding a comment here since it is non-idiomatic.